changelog/0037-upgrade-to-elasticsearch7.yml

```diff
@@ -0,0 +1,350 @@
+        ```
+        # to start user-pillow (replace with UserPillow if that's what you use)
+        cchq <env> service pillowtop start --only user-pillow
+        cchq <env> service pillowtop stop --only user-pillow
+        ```
+  3. At this point, your ES2 hosts should not be getting any HTTP requests.
+      - You can confirm this by doing a tcpdump on Elasticsearch port on any of your Elasticsearch 2 hosts
+        ```
+        tcpdump -A -s 0 'tcp port 9200 and (((ip[2:2] - ((ip[0]&0xf)<<2)) - ((tcp[12]&0xf0)>>2)) != 0)'_
+        ```
+  4. Reindex offset form/case data inserted after the Reindex phase.
+      ```
+      ./manage.py reindex_es_from_remote forms http://es2_cluster:9200 
+        --start_date=<last_reindex_date>
+      ./manage.py reindex_es_from_remote cases http://es2_cluster:9200 
+        --start_date=<last_reindex_date>
+      ```
+      - last_reindex_date is the date up to which you have already reindexed in the Reindex phase
+      - This and subsequent commands are to be run from the release directory that's pointing to Elasticsearch 7
+  5. Reindex rest of the indices by running below command for each of the index in `['users', 'domains', 'apps', 'groups', 'sms', 'report_cases', 'report_xforms', 'case_search']`
+      ```
+      ./manage.py reindex_es_from_remote <index_name> <es2_host>
+      ```
+      These can be run concurrently in different tmux/screen sessions
+  6. Reprocess form and case deletes since your first reindex from the reindex phase.
+      ```
+      ./manage.py reprocess_form_case_es_deletes --from=<date-of-when-migration-is-initiated>
+      ```
+  7. At this point, all of the Elasticsearch 2 data should be in Elasticsearch 7. You can confirm this by running the `reindex_es_from_remote` management command with `--print_index_size` option which displays number of documents in both the ES clusters. 
+  8. Run below command to update Elasticsearch settings on ES7 indices for usage
+      ```
+      ./manage.py initialize_es_indices --set-for-usage
+      ```
+  9. Update HQ services to point to Elasticsearch 7.
+      - Update below settings in you env's public.yml in addition to other setting changes made in git-upgrade-branch from Setup phase
+        ```
+        ELASTICSEARCH_MAJOR_VERSION: 7
+        ```
+      - Update config
+        ```
+        cchq <env> update-config --branch=<git-upgrade-branch>
+        ```
+      - Merge the git-upgrade-branch into master  
+  10. Make sure that HQ is able to connect to Elasticsearch 7
+      ```
+      cchq <env> django-manage check_services
+      ```
+  11. End downtime
+      ```
+      cchq <env> downtime end
+      ```
+  That's it, you are running HQ on Elasticsearch 7!
+
+  **Getting Estimate for switchover**
+
+  If you have large cluster you can get an estimate of downtime required for switchover. In most cases this is not necessary.
```

Comment:
Because, the Reindex phase would already have given good amount of insight into the durations and most envs would have smaller other indices (cases/forms are the only big indices usually, for solution envs others should be fairly small). For example, I am planning to skip this on ICDS.